### PR TITLE
Fetch / save bucket type properties

### PR DESCRIPTION
**Pull request by lukebakken**
[`CLIENTS-1074`](https://bashoeng.atlassian.net/browse/CLIENTS-1074)

### CODE REVIEW

**Comment by lukebakken** [comment]
@alexmoore I need to eliminate the copypasta in the store properties commands for buckets and bucket types, update some comments, then it'll be ready.

File: src/main/java/com/basho/riak/client/core/operations/StoreBucketPropsOperation.java [hunk 1]
```diff
@@ -15,36 +15,26 @@
...
 import com.basho.riak.protobuf.RiakPB;
 import com.google.protobuf.ByteString;
-import java.util.List;
 
 /**
  *
  * @author Brian Roach <roach at basho dot com>
  * @since 2.0
  */
 //TODO: return some sort of "success" instead of Void
```
Comment (alexmoore) [inline src/main/java/com/basho/riak/client/core/operations/StoreBucketPropsOperation.java]: Can probably remove this TODO.

File: src/main/java/com/basho/riak/client/api/commands/buckets/FetchBucketTypeProperties.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2013 Basho Technologies Inc
```
Comment (alexmoore) [inline src/main/java/com/basho/riak/client/api/commands/buckets/FetchBucketTypeProperties.java]: What year is it?!?

File: src/main/java/com/basho/riak/client/api/commands/buckets/StoreBucketTypeProperties.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2013 Basho Technologies Inc
```
Comment (alexmoore) [inline src/main/java/com/basho/riak/client/api/commands/buckets/StoreBucketTypeProperties.java]: Copy paste error

File: src/main/java/com/basho/riak/client/api/commands/buckets/StoreProperties.java [hunk 4]
```diff
@@ -0,0 +1,640 @@
+/*
+ * Copyright 2013 Basho Technologies Inc
```
Comment (alexmoore) [inline src/main/java/com/basho/riak/client/api/commands/buckets/StoreProperties.java]: Copy paste error

File: src/main/java/com/basho/riak/client/core/operations/FetchBucketTypePropsOperation.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2013 Basho Technologies Inc.
```
Comment (alexmoore) [inline src/main/java/com/basho/riak/client/core/operations/FetchBucketTypePropsOperation.java:2]: Copy paste error

File: src/test/java/com/basho/riak/client/core/operations/itest/ITestBucketTypeProperties.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2013 Basho Technologies Inc.
```
Comment (alexmoore) [inline src/test/java/com/basho/riak/client/core/operations/itest/ITestBucketTypeProperties.java]: Copy paste error

File: src/main/java/com/basho/riak/client/core/operations/StoreBucketTypePropsOperation.java [hunk 7]
```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2013 Basho Technologies Inc.
```
Comment (alexmoore) [inline src/main/java/com/basho/riak/client/core/operations/StoreBucketTypePropsOperation.java]: Copy paste error

**Review by alexmoore — APPROVED** [review]

File: src/main/java/com/basho/riak/client/core/operations/StoreBucketPropsOperation.java (hunk 1 above)
Comment (lukebakken) [reply src/main/java/com/basho/riak/client/core/operations/StoreBucketPropsOperation.java]: OK

File: src/main/java/com/basho/riak/client/api/commands/buckets/FetchBucketTypeProperties.java (hunk 2 above)
Comment (lukebakken) [reply src/main/java/com/basho/riak/client/api/commands/buckets/FetchBucketTypeProperties.java]: I'll remove the year on the files I added / modified.

Comment (christophermancini) [reply src/main/java/com/basho/riak/client/api/commands/buckets/FetchBucketTypeProperties.java]: Holy crap we went back in time?!?!?!

**Pull request merged**